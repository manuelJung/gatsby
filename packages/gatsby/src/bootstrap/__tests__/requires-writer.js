const { joinPath } = require(`gatsby-core-utils`)
const requiresWriter = require(`../requires-writer`)

const now = 1564229078922

const newMockState = () => {
  const pages = new Map()
  pages.set(`path1`, {
    component: `component1`,
    componentChunkName: `chunkName1`,
    matchPath: `matchPath1`,
    path: `/path1`,
  })
  pages.set(`path2`, {
    component: `component2`,
    componentChunkName: `chunkName2`,
    widgets: {
      WidgetA: `widget1`,
    },
    widgetChunkNames: {
      WidgetA: `widgetChunkName1`,
    },
    path: `/path2`,
  })
  const program = { directory: `/dir` }
  return { pages, program }
}

jest.mock(`fs-extra`, () => {
  return {
    writeFile: () => Promise.resolve(),
    move: () => {},
  }
})

const mockFsExtra = require(`fs-extra`)

describe(`requires-writer`, () => {
  beforeEach(() => {
    global.Date.now = () => now
    requiresWriter.resetLastHash()
  })
  describe(`writeAll`, () => {
    it(`writes requires files`, async () => {
      const spy = jest.spyOn(mockFsExtra, `writeFile`)
      await requiresWriter.writeAll(newMockState())
      expect(spy).toBeCalledWith(
        joinPath(`/dir`, `.cache`, `match-paths.json.${now}`),
        JSON.stringify([{ path: `/path1`, matchPath: `matchPath1` }], null, 4)
      )
      expect(spy).toMatchSnapshot()
    })
  })
})
