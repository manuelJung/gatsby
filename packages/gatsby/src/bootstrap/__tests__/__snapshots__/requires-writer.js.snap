// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requires-writer writeAll writes requires files 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/dir/.cache/sync-requires.js.1564229078922",
      "const { hot } = require(\\"react-hot-loader/root\\")

// prefer default export if available
const preferDefault = m => m && m.default || m


exports.components = {
  \\"chunkName1\\": hot(preferDefault(require(\\"component1\\"))),
  \\"chunkName2\\": hot(preferDefault(require(\\"component2\\"))),
  \\"widgetChunkName1\\": hot(preferDefault(require(\\"widget1\\")))
}

",
    ],
    Array [
      "/dir/.cache/async-requires.js.1564229078922",
      "// prefer default export if available
const preferDefault = m => m && m.default || m

exports.components = {
  \\"chunkName1\\": () => import(\\"component1\\" /* webpackChunkName: \\"chunkName1\\" */),
  \\"chunkName2\\": () => import(\\"component2\\" /* webpackChunkName: \\"chunkName2\\" */),
  \\"widgetChunkName1\\": () => import(\\"widget1\\" /* webpackChunkName: \\"widgetChunkName1\\" */)
}

",
    ],
    Array [
      "/dir/.cache/match-paths.json.1564229078922",
      "[
    {
        \\"path\\": \\"/path1\\",
        \\"matchPath\\": \\"matchPath1\\"
    }
]",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
